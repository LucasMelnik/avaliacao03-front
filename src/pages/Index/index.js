import React, { useEffect, useState } from "react";

import api from '../../api/index'

export default function Index() {

    const [contacts, setContacts] = useState([])

    async function handleRender() {
        const response = await api.get('/contacts')

        setContacts(response.data)

        console.log(contacts)
    }

    useEffect(() => {
        handleRender();
    }, []);

    return(
        <>
            <table>
                <tr>
                    <th>Id</th>
                    <th>Name</th>
                    <th>E-mail</th>
                    <th>Phone</th>
                </tr>
                {contacts.map((contact) => (
                    <tr>
                        <td>{contact.id}</td>
                        <td>{contact.name}</td>
                        <td>{contact.email}</td>
                        <td>{contact.phone}</td>
                    </tr>
                ))}
            </table>
        </>
    )
}