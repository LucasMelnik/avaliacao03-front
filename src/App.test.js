import React from "react"
import { shallow, } from 'enzyme'
import '@testing-library/jest-dom/extend-expect'
import '@testing-library/jest-dom'
import Index from './pages/Index/index'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe("Testing Index Component", () => {
    it("should render correctly", () => {
        const wrapper = shallow(
            <Index />
        )

        expect(wrapper).toMatchSnapshot();
    })
    
})