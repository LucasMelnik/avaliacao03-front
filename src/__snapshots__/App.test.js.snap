// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Index Component should render correctly 1`] = `ShallowWrapper {}`;
